File: 2249-count-the-hidden-sequences/2249-count-the-hidden-sequences.py
class Solution:
    def numberOfArrays(self, differences: List[int], lower: int, upper: int) -> int:
        curr, mn, mx = 0, 0, 0
        for d in differences:
            curr += d
            mn = min(mn, curr)
            mx = max(mx, curr)
            if (upper - mx) - (lower - mn) + 1 <= 0: return 0
        
        return upper - mx - lower + mn + 1

File: 2249-count-the-hidden-sequences/2249-count-the-hidden-sequences.go
func numberOfArrays(differences []int, lower int, upper int) int {
    curr, min, max := 0, 0, 0
    for _, d := range differences {
        curr += d
        if curr < min {
            min = curr
        }
        if curr > max {
            max = curr
        }

        if (upper - max) - (lower - min) + 1 <= 0 {
            return 0
        }
    }
    return (upper - lower) - (max - min) + 1
}